network/src/tests/mod.rs

```diff
@@ -20,20 +20,98 @@ use super::{PolkadotProtocol, Status, Message, FullStatus};
 use validation::SessionParams;
 
 use polkadot_validation::GenericStatement;
-use polkadot_primitives::{Block, Hash, SessionKey};
+use polkadot_primitives::SessionKey;
 use polkadot_primitives::parachain::{CandidateReceipt, HeadData, BlockData, CollatorId, ValidatorId};
+use sr_primitives::Justification;
+use sr_primitives::generic::BlockId;
+use sr_primitives::traits::NumberFor;
+use substrate_primitives::H256;
 use substrate_primitives::crypto::UncheckedInto;
 use codec::Encode;
+use substrate_consensus::import_queue::{SharedBlockImport, SharedJustificationImport, Verifier};
+use substrate_consensus::{Error as ConsensusError, ErrorKind as ConsensusErrorKind, JustificationImport};
 use substrate_network::{
-	Severity, PeerId, PeerInfo, ClientHandle, Context, config::Roles,
+	Severity, PeerId, PeerInfo, ClientHandle, Context, config::ProtocolConfig, config::Roles,
 	message::Message as SubstrateMessage, specialization::NetworkSpecialization,
 	generic_message::Message as GenericMessage
 };
+use substrate_network::test::{
+	Block, DummySpecialization, Hash, TestNet, TestNetFactory,
+	PassThroughVerifier, Peer, PeersClient, SpecializationFactory
+};
 
 use futures::Future;
+use std::sync::Arc;
 
 mod validation;
 
+impl SpecializationFactory for PolkadotProtocol {
+	fn create() -> PolkadotProtocol {
+		PolkadotProtocol::new(None)
+	}
+}
+
+pub struct ForceFinalized(Arc<PeersClient>);
+
+impl JustificationImport<Block> for ForceFinalized {
+	type Error = ConsensusError;
+
+	fn import_justification(
```

Comment:
these arguments could fit on a single line (it's no longer than https://github.com/paritytech/polkadot/pull/202/files#diff-383209c932e7187683ba4eea46d5b30dR95)